src/python/pants/binaries/binary_tool.py

```diff
@@ -50,6 +50,8 @@ class BinaryToolBase(Subsystem):
   # Subclasses may set this to provide extra register() kwargs for the --version option.
   extra_version_option_kwargs = None
 
+  allow_version_override = True
```

Comment:
I might maybe just change this to a `@classproperty` named something like `hardcoded_version` which defaults to `None`? That would mean the zinc native-image factory subsystem could just set `hardcoded_version` instead of having to override `version()`.

Reply:
I agree with both of your design suggestions here.

On the other hand, I think that the underlying idea that we won't want to be able to point this at multiple versions is not accurate. It's fine to change the default, but I don't think that we want to hardcode it.

Reply:
I pretty strongly think it's important that there's one way to select what version of zinc is being used; I would be very confused, both as a pants user and as someone supporting other pants users, trying to work out which version settings override which...

Reply:
I don't see `allow_version_override` as being any less confusing than something like `hardcoded_version`, but I do see it being more indirect (where is the actual override occurring?). I misinterpreted this reading it earlier as assuming that `allow_version_override` meant "allow the class to override any version set by the user", as opposed to the other way around -- because I didn't think of the user specifying a specific `--version` option value as "overriding" the version, but rather just "setting" it.

Reply:
> On the other hand, I think that the underlying idea that we won't want to be able to point this at multiple versions is not accurate. It's fine to change the default, but I don't think that we want to hardcode it.

I don't think we'll want to provide native images as BinaryTools for very long, though -- e.g. #6893 will be able to avoid any hardcoded dependence on a specific zinc or scala-library version by just converting whatever the input classpath ends up being into an image! So I would recommend shaping the solution in a way that lets us remove as cleanly as possible later (both of the approaches here would seem to work).

Reply:
Yep. This patch represents a temporary solution to allow us to iterate. And we're immediately going to have multiple versions of the native image.